scylla/tests/lwt_optimisation.rs

```diff
@@ -0,0 +1,149 @@
+mod utils;
+
+use scylla::frame::types;
+use scylla::retry_policy::FallthroughRetryPolicy;
+use scylla::transport::session::Session;
+use scylla::SessionBuilder;
+use scylla::{frame::protocol_features::ProtocolFeatures, test_utils::unique_keyspace_name};
+use std::sync::Arc;
+use tokio::sync::mpsc;
+use utils::test_with_3_node_cluster;
+
+use scylla_proxy::{
+    Condition, ProxyError, Reaction, RequestFrame, RequestOpcode, RequestReaction, RequestRule,
+    ResponseOpcode, ResponseReaction, ResponseRule, ShardAwareness, WorkerError,
+};
+
+#[tokio::test]
+#[ntest::timeout(20000)]
+async fn if_lwt_optimisation_mark_offered_then_negotiatied_and_lwt_routed_optimally() {
+    // This is just to increase the likelyhood that only intended prepared statements (which contain this mark) are captures by the proxy.
+    const MAGIC_MARK: i32 = 123;
+
+    // We choose Unaware to support testing Cassandra
```

Comment:
Our "cassandra" check in CI sets up only one Cassamdra node as far as I know.

Reply:
That's true, but one can run this test on a multi-node Cassandra cluster and verify that everything works.

Reply:
In that case we need to extend the CI to set up three nodes instead of one. You seem to already have done some improvements towards Cassandra support both in the proxy and in the "cassandra" CI check, so why not finish this?

It would probably be the best to extract all the improvements related to Cassandra + proxy support to a separate PR.